src/views/edit-page.vue

```diff
@@ -0,0 +1,92 @@
+<template>
+  <Header />
+  <form id="updatePost" @submit.prevent="applyEditOnNewsFeed">
+    <div class="input-group mb-3 title">
+      <span class="input-group-text" id="basic-addon1">Title</span>
+      <input
+        type="text"
+        class="form-control"
+        aria-label="title"
+        aria-describedby="basic-addon1"
+        v-model="post.title"
+      />
+    </div>
+    <div class="input-group">
+      <span class="input-group-text">Message</span>
+      <textarea
+        class="form-control message"
+        aria-label="With textarea"
+        v-model="post.message"
+      ></textarea>
+    </div>
+    <button
+      class="btn btn-primary"
+      type="submit"
+      @click.prevent="applyEditOnNewsFeed(post)"
+    >
+      Edit
+    </button>
+  </form>
+</template>
+
+<script lang="ts">
+import { defineComponent } from "vue";
+import INewsFeed from "@/interface/news-feed.interface";
+import Header from "@/shared/component/header.vue";
+import { useRouter, useRoute } from "vue-router";
+import RouteName from "@/enum/routes-name.enum";
+import useGetById from "@/composables/feed/useGetById";
+import useEditById from "@/composables/feed/useEditbyId";
+
+export default defineComponent({
+  name: "Edit",
+  components: {
+    Header,
+  },
+  setup() {
+    const id = parseInt(useRoute().params.id as string);
+
+    const { post } = useGetById(id);
+
+    const date = new Date();
+
+    const router = useRouter();
+
+    async function applyEditOnNewsFeed(editedPost: INewsFeed) {
+      editedPost.updatedAt = date.toLocaleDateString();
+      const { res } = useEditById(editedPost);
+      if (res) {
+        alert("have been updated");
+        router.push({
+          name: RouteName.FEED,
+        });
+      } else {
+        alert("update failed edit");
+      }
+    }
+    return { post, router, applyEditOnNewsFeed };
```

Comment:
i looks like router is not used in the template, why expose it?

Reply:
deleted router since it is not used in the template.